components/tournaments_card/commons/tournaments_card_base.lua

```diff
@@ -0,0 +1,659 @@
+	end
+
+	return table.concat(locations)
+end
+
+function BaseTournamentsCard._displayLocation(locationData, locationIndex)
+	local display = ''
+	local region = locationData['region' .. locationIndex]
+	local country = locationData['country' .. locationIndex]
+	local city = locationData['city' .. locationIndex]
+
+	if country then
+		display = Flags.Icon{flag = country} .. NONBREAKING_SPACE
+	elseif city and region then
+		display = Flags.Icon{flag = region} .. NONBREAKING_SPACE
+	elseif region then
+		return Region.run{region = region, onlyDisplay = true}
+	end
+
+	return String.nilIfEmpty(display .. (city or Flags.CountryName(country)))
+end
+
+function BaseTournamentsCard._dateDisplay(startDate, endDate, status)
+	if status == POSTPONED or status == DELAYED then
+		return 'Postponed'
+	end
+
+	if startDate == endDate then
+		return LANG:formatDate('M j, Y', startDate)
+	end
+
+	local startYear, startMonth = startDate:match('(%d+)-(%d+)-%d+')
+	local endYear, endMonth = endDate:match('(%d+)-(%d+)-%d+')
+
+	if startYear ~= endYear then
+		return LANG:formatDate('M j, Y', startDate) .. ' - ' .. LANG:formatDate('M j, Y', endDate)
+	end
+
+	if startMonth == endMonth then
+		return LANG:formatDate('M j', startDate) .. ' - ' .. LANG:formatDate('j, Y', endDate)
+	end
+
+	return LANG:formatDate('M j', startDate) .. ' - ' .. LANG:formatDate('M j, Y', endDate)
+end
+
+function BaseTournamentsCard:_fetchPlacementData(tournamentData)
+	local placements = {}
+
+	local queryData = mw.ext.LiquipediaDB.lpdb('placement', {
+		conditions = self:_buildPlacementConditions(tournamentData),
+		query = 'opponentname, opponenttype, opponenttemplate, opponentplayers, placement, extradata',
+		order = 'placement asc',
+		limit = 50,
+	})
+
+	if self.config.showQualifierColumnOverWinnerRunnerup then
+		if Table.isEmpty(queryData) then
+			return {qualified = {Opponent.tbd(Opponent.team)}}
+		end
+		return {qualified = queryData}
+	end
+
+	for _, item in ipairs(queryData) do
+		local place = tonumber(mw.text.split(item.placement, '-')[1])
+		if not place and item.placement == 'W' then
+			place = 1
+		elseif not place and item.placement == 'L' then
+			place = 2
+		end
+
+		if place then
+			if not placements[place] then
+				placements[place] = {}
+			end
+
+			local opponent = Opponent.fromLpdbStruct(item)
+			if not opponent then
+				mw.logObject({pageName = tournamentData.pagename, place = item.placement}, 'Invalid Prize Pool Data returned from')
+			end
+			if Opponent.isTbd(opponent) then
+				opponent = Opponent.tbd(Opponent.team)
+			end
+			table.insert(placements[place], opponent)
+		end
+	end
+
+	if Table.isEmpty(placements[1]) then
+		placements[1] = {Opponent.tbd(Opponent.team)}
+	end
+
+	if Table.isEmpty(placements[2]) then
+		placements[2] = {Opponent.tbd(Opponent.team)}
+	end
+
+	return placements
+end
+
+function BaseTournamentsCard:_buildPlacementConditions(tournamentData)
```

Comment:
Does awards need to be excluded? Don't recall exactly how their placement values are done.

Reply:
it factually does via `config.allowedPlacements` (since awards do not have a set placement.placement)
but if wanted we can add it explicitly